tests/baselines/reference/controlFlowAssignmentPatternOrder.errors.txt

```diff
@@ -0,0 +1,97 @@
+controlFlowAssignmentPatternOrder.ts(6,9): error TS2339: Property '1' does not exist on type 'never'.
+controlFlowAssignmentPatternOrder.ts(12,9): error TS2339: Property '0' does not exist on type 'never'.
+controlFlowAssignmentPatternOrder.ts(31,9): error TS2339: Property '1' does not exist on type 'never'.
+controlFlowAssignmentPatternOrder.ts(37,9): error TS2339: Property '0' does not exist on type 'never'.
+controlFlowAssignmentPatternOrder.ts(56,14): error TS2339: Property '1' does not exist on type 'never'.
+controlFlowAssignmentPatternOrder.ts(62,14): error TS2339: Property '0' does not exist on type 'never'.
+
+
+==== controlFlowAssignmentPatternOrder.ts (6 errors) ====
+    // https://github.com/microsoft/TypeScript/pull/41094#issuecomment-716044363
+    declare function f(): void;
+    {
+        let a: 0 | 1 = 0;
+        let b: 0 | 1 | 9;
+        [{ [(a = 1)]: b } = [9, a] as const] = [];
+            ~~~~~~~
+!!! error TS2339: Property '1' does not exist on type 'never'.

```

Comment:
I'm not entirely sure if the error here is completely desirable but it feels consistent to me with things like:
```ts
const obj = { b: '' }
let a: number
;({ a = 10 } = obj) // Property 'a' does not exist on type '{ b: string; }'.(2339)
```

What I mean is that despite the first element in this array pattern having an initializer it's the "right-side" type of the original source that is used to check for valid property accesses. 

So since we have `never[]` on the right side, its first element is actually `never` and that has no properties so disallowing those patterns here looks desirable to me.

Reply:
This error doesn't look right to me: it says "Property '1' does not exist on type 'never'", but we should be getting property `1` of type `[9, typeof a]`, and that tuple should not have type `never`.